cmd/agent/service_windows.go

```diff
@@ -0,0 +1,81 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	util_log "github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/config"
+	"log"
+	"os"
+	"time"
+
+	"golang.org/x/sys/windows/svc"
+)
+
+const ServiceName = "Grafana Cloud Agent"
+const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+
+type AgentService struct{}
+
+func (m *AgentService) Execute(args []string, serviceRequests <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	changes <- svc.Status{State: svc.StartPending}
+
+	// Executable name and any command line parameters will be placed into os.args, this comes from
+	// registry key `Computer\HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Services\<servicename>\ImagePath`
+	// oddly enough args is blank
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	cfg, err := config.Load(fs, os.Args[1:])
+	if err != nil {
+		log.Fatalln(err)
+	}
+	// After this point we can use util_log.Logger and stop using the log package
+	util_log.InitLogger(&cfg.Server)
+	logger := util_log.Logger
+
+	srv, err := NewAgentServer(logger, cfg)
+	if err != nil {
+		level.Error(logger).Log("msg", "error starting agent", "err", err)
+		os.Exit(1)
+	}
+	// Kick off the server in the background so that we can respond to status queries
+	go srv.srv.Run()
```

Comment:
If this exits we need to stop the for loop below, can you add a channel here to detect?

Reply:
Please take a look at what I did to ensure it looks right, I tested channel features in a test project to make sure my understanding was right.